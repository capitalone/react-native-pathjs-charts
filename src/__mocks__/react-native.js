import React from 'react'

export const View = ({ children }) => {
  return (
    <div>{children}</div>
  )
}
