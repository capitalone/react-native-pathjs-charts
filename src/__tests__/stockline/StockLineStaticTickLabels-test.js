import 'react-native'
import React from 'react'
import StockLine from '../../StockLine'
import renderer from 'react-test-renderer'

let data = [
  [{
    "x": 0,
    "y": 47782
  }, {
    "x": 1,
    "y": 48497
  }, {
    "x": 2,
    "y": 77128
  }, {
    "x": 3,
    "y": 73413
  }, {
    "x": 4,
    "y": 58257
  }, {
    "x": 5,
    "y": 40579
  }, {
    "x": 6,
    "y": 72893
  }]
]
let options = {
  width: 250,
  height: 250,
  color: '#2980B9',
  margin: {
    top: 10,
    left: 35,
    bottom: 30,
    right: 10
  },
  animate: {
    type: 'delayed',
    duration: 200
  },
  axisX: {
    showAxis: true,
    showLines: true,
    showLabels: true,
    showTicks: true,
    zeroAxis: false,
    orient: 'bottom',
    tickValues: [
      {value:'name1'},
      {value:'name2'},
      {value:'name3'},
      {value:'name4'},
      {value:'name5'},
      {value:'name6'},
      {value:'name7'}
    ],
    label: {
      fontFamily: 'Arial',
      fontSize: 8,
      fontWeight: true,
      fill: '#34495E'
    }
  },
  axisY: {
    showAxis: true,
    showLines: true,
    showLabels: true,
    showTicks: true,
    zeroAxis: false,
    orient: 'left',
    tickValues: [],
    label: {
      fontFamily: 'Arial',
      fontSize: 8,
      fontWeight: true,
      fill: '#34495E'
    }
  }
}

it('renders an example chart correctly', () => {
  let tree = renderer.create(
    <StockLine data={data}
      options={options}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
