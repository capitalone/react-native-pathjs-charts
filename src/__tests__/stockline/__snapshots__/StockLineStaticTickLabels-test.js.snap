// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an example chart correctly 1`] = `
<view
  height={290}
  mockedComponent="svg-component"
  width={295}
>
  <view
    mockedComponent="svg-G"
    x={35}
    y={10}
  >
    <view
      mockedComponent="svg-G"
      x={-35}
      y={-10}
    >
      <view
        d="M NaN NaN L NaN NaN"
        fill="none"
        mockedComponent="svg-Path"
        stroke="#3E90F0"
        strokeOpacity={0.5}
      />
      <view
        d="M NaN NaN L NaN NaN"
        fill="none"
        mockedComponent="svg-Path"
        stroke="#3E90F0"
        strokeOpacity={0.5}
      />
      <view
        d="M NaN NaN L NaN NaN"
        fill="none"
        mockedComponent="svg-Path"
        stroke="#3E90F0"
        strokeOpacity={0.5}
      />
      <view
        d="M NaN NaN L NaN NaN"
        fill="none"
        mockedComponent="svg-Path"
        stroke="#3E90F0"
        strokeOpacity={0.5}
      />
      <view
        d="M NaN NaN L NaN NaN"
        fill="none"
        mockedComponent="svg-Path"
        stroke="#3E90F0"
        strokeOpacity={0.5}
      />
      <view
        d="M NaN NaN L NaN NaN"
        fill="none"
        mockedComponent="svg-Path"
        stroke="#3E90F0"
        strokeOpacity={0.5}
      />
      <view
        d="M NaN NaN L NaN NaN"
        fill="none"
        mockedComponent="svg-Path"
        stroke="#3E90F0"
        strokeOpacity={0.5}
      />
    </view>
    <view
      mockedComponent="svg-G"
      x={-35}
      y={-10}
    />
    <view
      d="M NaN NaN L NaN NaN L NaN NaN L NaN NaN L NaN NaN L NaN NaN L NaN NaN L NaN NaN L NaN NaN Z "
      fill="#3199de"
      fillOpacity={0.5}
      mockedComponent="svg-Path"
      stroke="none"
    />
    <view
      d="M NaN NaN L NaN NaN L NaN NaN L NaN NaN L NaN NaN L NaN NaN L NaN NaN"
      fill="none"
      mockedComponent="svg-Path"
      stroke="#3199de"
      strokeDasharray={Array []}
      strokeOpacity={1}
      strokeWidth="1"
    />
    <view
      mockedComponent="svg-G"
    >
      <view
        mockedComponent="svg-G"
        x={-35}
        y={-10}
      >
        <view
          d="M NaN NaN L NaN NaN Z "
          fill="none"
          mockedComponent="svg-Path"
          stroke="#3E90F0"
          strokeOpacity={0.5}
          strokeWidth={3}
        />
      </view>
      <view
        mockedComponent="svg-G"
        x={NaN}
        y={NaN}
      >
        <view
          cx="0"
          cy="0"
          fill="grey"
          mockedComponent="svg-Circle"
          r={2}
          stroke="grey"
        />
        <view
          fill="#34495E"
          fontFamily="Arial"
          fontSize={8}
          fontStyle="normal"
          fontWeight="bold"
          mockedComponent="svg-Text"
          textAnchor="middle"
          x={0}
          y={5}
        >
          name1
        </view>
      </view>
      <view
        mockedComponent="svg-G"
        x={NaN}
        y={NaN}
      >
        <view
          cx="0"
          cy="0"
          fill="grey"
          mockedComponent="svg-Circle"
          r={2}
          stroke="grey"
        />
        <view
          fill="#34495E"
          fontFamily="Arial"
          fontSize={8}
          fontStyle="normal"
          fontWeight="bold"
          mockedComponent="svg-Text"
          textAnchor="middle"
          x={0}
          y={5}
        >
          name2
        </view>
      </view>
      <view
        mockedComponent="svg-G"
        x={NaN}
        y={NaN}
      >
        <view
          cx="0"
          cy="0"
          fill="grey"
          mockedComponent="svg-Circle"
          r={2}
          stroke="grey"
        />
        <view
          fill="#34495E"
          fontFamily="Arial"
          fontSize={8}
          fontStyle="normal"
          fontWeight="bold"
          mockedComponent="svg-Text"
          textAnchor="middle"
          x={0}
          y={5}
        >
          name3
        </view>
      </view>
      <view
        mockedComponent="svg-G"
        x={NaN}
        y={NaN}
      >
        <view
          cx="0"
          cy="0"
          fill="grey"
          mockedComponent="svg-Circle"
          r={2}
          stroke="grey"
        />
        <view
          fill="#34495E"
          fontFamily="Arial"
          fontSize={8}
          fontStyle="normal"
          fontWeight="bold"
          mockedComponent="svg-Text"
          textAnchor="middle"
          x={0}
          y={5}
        >
          name4
        </view>
      </view>
      <view
        mockedComponent="svg-G"
        x={NaN}
        y={NaN}
      >
        <view
          cx="0"
          cy="0"
          fill="grey"
          mockedComponent="svg-Circle"
          r={2}
          stroke="grey"
        />
        <view
          fill="#34495E"
          fontFamily="Arial"
          fontSize={8}
          fontStyle="normal"
          fontWeight="bold"
          mockedComponent="svg-Text"
          textAnchor="middle"
          x={0}
          y={5}
        >
          name5
        </view>
      </view>
      <view
        mockedComponent="svg-G"
        x={NaN}
        y={NaN}
      >
        <view
          cx="0"
          cy="0"
          fill="grey"
          mockedComponent="svg-Circle"
          r={2}
          stroke="grey"
        />
        <view
          fill="#34495E"
          fontFamily="Arial"
          fontSize={8}
          fontStyle="normal"
          fontWeight="bold"
          mockedComponent="svg-Text"
          textAnchor="middle"
          x={0}
          y={5}
        >
          name6
        </view>
      </view>
      <view
        mockedComponent="svg-G"
        x={NaN}
        y={NaN}
      >
        <view
          cx="0"
          cy="0"
          fill="grey"
          mockedComponent="svg-Circle"
          r={2}
          stroke="grey"
        />
        <view
          fill="#34495E"
          fontFamily="Arial"
          fontSize={8}
          fontStyle="normal"
          fontWeight="bold"
          mockedComponent="svg-Text"
          textAnchor="middle"
          x={0}
          y={5}
        >
          name7
        </view>
      </view>
    </view>
    <view
      mockedComponent="svg-G"
    >
      <view
        mockedComponent="svg-G"
        x={-35}
        y={-10}
      >
        <view
          d="M NaN NaN L NaN NaN Z "
          fill="none"
          mockedComponent="svg-Path"
          stroke="#3E90F0"
          strokeOpacity={0.5}
          strokeWidth={3}
        />
      </view>
    </view>
  </view>
</view>
`;
